heat/core/dndarray.py

```diff
@@ -1399,222 +1401,175 @@ def __getitem__(self, key):
-                            key[self.split] = key[self.split] - chunk_start
-                            arr = self.__array[tuple(key)]
-                            gout = list(arr.shape)
-                    if 0 in arr.shape:
-                        # arr is empty
-                        # gout is all 0s and is the proper shape
-                        warnings.warn(
-                            "This process (rank: {}) is without data after slicing, running the .balance_() function is recommended".format(
-                                self.comm.rank
-                            ),
-                            ResourceWarning,
-                        )
-
-            # if the given axes are only a slice
-            elif isinstance(key, slice) and self.split == 0:
-                gout = [0] * len(self.gshape)
-                # reduce the dims if the slices are only one element in length
-                start = key.start if key.start is not None else 0
-                stop = key.stop if key.stop is not None else self.gshape[0]
-                step = key.step if key.step is not None else 1
+                return DNDarray(
+                    self.__array[key], gout, self.dtype, new_split, self.device, self.comm
+                )
 
-                if self.split >= len(gout):
-                    new_split = len(gout) - 1 if len(gout) - 1 > 0 else 0
-                else:
-                    new_split = self.split
-                key_set = set(range(start, stop, step))
-                overlap = list(key_set & chunk_set)
-                if overlap:
-                    overlap.sort()
-                    hold = [x - chunk_start for x in overlap]
-                    key = slice(min(hold), max(hold) + 1, step)
-                    arr = self.__array[key]
-                    gout = list(arr.shape)
-                else:
-                    warnings.warn(
-                        "This process (rank: {}) is without data after slicing, "
-                        "running the .balance_() function is recommended".format(self.comm.rank),
-                        ResourceWarning,
+        else:
+            rank = self.comm.rank
+            ends = []
+            for pr in range(self.comm.size):
+                _, _, e = self.comm.chunk(self.shape, self.split, rank=pr)
+                ends.append(e[self.split].stop - e[self.split].start)
+            ends = torch.tensor(ends, device=self.device.torch_device)
+            chunk_ends = ends.cumsum(dim=0)
+            chunk_starts = torch.tensor([0] + chunk_ends.tolist(), device=self.device.torch_device)
+            chunk_start = chunk_starts[rank]
+            chunk_end = chunk_ends[rank]
+            arr = torch.Tensor()
+            # all keys should be tuples here
+            gout = [0] * len(self.gshape)
+            # handle the dimensional reduction for integers
+            ints = sum([isinstance(it, int) for it in key])
```

Comment:
probably this should be an all() instead of sum(), also: remove the square brackets and transform into a generator

Reply:
in this case, the number of integers in the key is used to determine the new split axis so it as to be the sum. but the square brackets were removed